pkg/controller/azure/compute/akscluster.go

```diff
@@ -315,18 +326,28 @@ func (r *Reconciler) delete(instance *computev1alpha1.AKSCluster, aksClient *azu
 		deleteFutureJSON, _ := deleteFuture.MarshalJSON()
 		log.V(logging.Debug).Info("started delete of AKS cluster", "instance", instance, "operation", string(deleteFutureJSON))
 
+		// delete the role assignment if created
+		if instance.Spec.VnetSubnetID != "" {
+			log.V(logging.Debug).Info("deleting role assignment for service principal for AKS cluster", "instance", instance)
+			name := string(instance.GetUID())
+			err = aksClient.RoleAssignmentsAPI.DeleteRoleAssignment(ctx, instance.Spec.VnetSubnetID, name)
```

Comment:
we create the RoleAssignment with `instance.Status.ServicePrincipalID, instance.Spec.VnetSubnetID` and we delete it with only `instance.Spec.VnetSubnetID`. The delete call doesn't also need the Service Principal ID to properly identify the role assignment?

Reply:
It does not. Because the role assignment must have a unique name, it is able to be identified without including the service principal.